### PR TITLE
opensync: remove stale clients

### PR DESCRIPTION
**Pull request by ArifAlam**
Remove stale clients from client list in case disconnect event was
not received.

Signed-off-by: Arif Alam <arif.alam@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c [hunk 1]
```diff
@@ -227,6 +233,26 @@ static void nl80211_del_station(struct nlattr **tb, char *ifname)
...
+	time_t timeout = 0, t = 0;
+
+	avl_for_each_element_safe(&sta_tree, sta, avl, tmp) {
+		timeout = sta->last_update + (EVSCHED_SEC(STA_POLL_INTERVAL*2));
+		t = time(NULL);
+		if (t > timeout) {
+			print_mac(mac, sta->addr);
+			LOGN("%s: Remove stale STA %s %ld %ld %ld", __func__, mac,
+					t, timeout, sta->last_update);
+			vif_del_sta_rate_rule(sta->addr, ifname);
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c:249]: This sequence of three calls is also made in nl80211_del_station().  Maybe it should be a little helper function.  If more operations are later tied to this data structure, then we eliminate the risk of them only being called in one code path.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c [hunk 2]
```diff
@@ -227,6 +233,26 @@ static void nl80211_del_station(struct nlattr **tb, char *ifname)
...
 }
 
+static void remove_stale_station(char *ifname)
+{
+	struct wifi_station *sta = NULL, *tmp = NULL;
+	char mac[ETH_ALEN * 3];
+	time_t timeout = 0, t = 0;
+
+	avl_for_each_element_safe(&sta_tree, sta, avl, tmp) {
+		timeout = sta->last_update + (EVSCHED_SEC(STA_POLL_INTERVAL*2));
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c:243]: How did you choose two missed polls?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c [hunk 3]
```diff
@@ -479,11 +505,12 @@ static void vif_poll_stations(void *arg)
 	avl_for_each_element(&wif_tree, wif, avl) {
 		struct nl_msg *msg;
 
+		remove_stale_station(wif->name);
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c:508]: The call to remove_stale_station() could be done after the unl_genl_request().  This would avoid "deletion and then immediate reinsertion" corner case.

File: feeds/wlan-ap/opensync/src/platform/openwrt/rootfs/common/etc/init.d/opensync [hunk 4]
```diff
@@ -47,5 +47,5 @@ stop_service() {
     echo "Removing certificates"
     rm -rf ${CERTS_DEST_PATH}
     echo "Killing managers"
-    killall dm wm sm cm nm lm qm cmdm rrm um uccm
+    killall -s SIGKILL dm wm sm cm nm lm qm cmdm rrm um uccm
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/rootfs/common/etc/init.d/opensync:50]: Unrelated.  Should be a separate commit.

**Review by RickSommerville — CHANGES_REQUESTED** [review]

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c [hunk 5]
```diff
@@ -169,6 +174,7 @@ static void nl80211_add_station(struct nlattr **tb, char *ifname)
 	struct wifi_iface *wif;
 	uint8_t *addr;
 
+
```
[the remark this answers is not in the document]
Comment (ArifAlam) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c]: Done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c (hunk 1 above)
Comment (ArifAlam) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c]: Done. Moved to _nl80211_del_station.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c (hunk 3 above)
Comment (ArifAlam) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c]: Done.

File: feeds/wlan-ap/opensync/src/platform/openwrt/rootfs/common/etc/init.d/opensync (hunk 4 above)
Comment (ArifAlam) [reply feeds/wlan-ap/opensync/src/platform/openwrt/rootfs/common/etc/init.d/opensync]: Done.  This change is in a separate PR.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c (hunk 2 above)
Comment (ArifAlam) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c:241]: Just being extra cautious.

**Comment by RickSommerville** [comment]
Committed by git am
